lib/strings.nix

```diff
@@ -703,7 +703,7 @@ rec {
   */
   getName = x:
    let
-     parse = drv: (parseDrvName drv).name;
+     parse = drv: (parseDrvName (unsafeDiscardStringContext drv)).name;
```

Comment:
Should have a test case and justification in the code for why this is necessary and safe

Reply:
Continued in https://github.com/NixOS/nixpkgs/pull/177272#discussion_r1328114168